server/src/main/java/org/elasticsearch/rest/Scope.java

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.rest;
+
+public enum Scope {
+    PUBLIC, // available to all requests
+    INTERNAL // available only to requests with a X-elastic-internal-origin header
+}
```

Comment:
The rest API spec encodes `Visibility` which is eerily similar to `Scope` 

https://github.com/elastic/elasticsearch/blob/d27102fa7e8d2f47ec7c3e6bcf5da969a0823603/test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestApi.java#L46-L50

Should we look into merging the two concepts? 

See: https://github.com/elastic/elasticsearch/pull/56104

cc @sethmlarson

Reply:
I think this is slightly different, in that Scope here is specifically about serverless, while Visibility applies to self managed ES. The name "scope" should probably be more specific to make that clear.